controllers/controllers/workloads/cfprocess_controller.go

```diff
@@ -165,7 +180,9 @@ func (r *CFProcessReconciler) cleanUpAppWorkloads(ctx context.Context, cfProcess
 	}
 
 	for i, currentAppWorkload := range appWorkloadsForProcess {
-		if desiredState == korifiv1alpha1.StoppedState || currentAppWorkload.Labels[korifiv1alpha1.CFAppRevisionKey] != cfAppRev {
+		if desiredState == korifiv1alpha1.StoppedState ||
+			desiredProcessInstances(cfProcess) == 0 ||
+			currentAppWorkload.Labels[korifiv1alpha1.CFAppRevisionKey] != cfAppRev {
```

Comment:
Maybe these conditions can be extracted in a predicate function `needsCleanup`, so that it is symmetrical with the creation case?